src/index.d.ts

```diff
@@ -58,6 +58,15 @@ export function inject<S, P, I, C>(
 // ):
 // <T extends IReactComponent<P & S>(target: T) => IReactComponent<P> & IWrappedComponent<T>
 
+/**
+ * disposeOnUnmount
+ */
+export function disposeOnUnmount(target: React.Component, propertyKey: string): void
+export function disposeOnUnmount<TF extends Function | Function[]>(
```

Comment:
I would explicitly type `Function` as `Disposer`, where `Disposer` is `() => void`

Reply:
will do